language/move-binary-format/src/compatibility.rs

```diff
@@ -58,27 +54,34 @@ impl CompatibilityConfig {
         }
     }
 
+    pub fn new(
```

Comment:
No need to fix in this PR, but might be worth adding a TODO to expose a no-arg entrypoint for each valid config rather than a single entrypoint with 3 bools. That will let us eliminate the `assert` below (since we simplfy won't expose an entrypoint that allows `check_friend_linking` without `check_struct_and_function_linking`.

Reply:
How about we change `check_struct_and_function_linking` to `check_struct_and_pub_function_linking`, so `check_struct_and_pub_function_linking` +  `check_friend_linking` = origin `check_struct_and_function_linking`.

And `new(check_struct_and_pub_function_linking:false, check_friend_linking: true)` seem be useless(maybe for only check the `friend`?), but the logic is valid.

Reply:
I do the change, and let `check_struct_and_pub_function_linking` and `check_friend_linking` no overlap.